### PR TITLE
Sequencer cleanup

### PR DESCRIPTION
**Pull request by eljeff**
Make things more consistent and deprecate old functions in favor of direct track access

### CODE REVIEW

**Review by aure — COMMENTED** [review]
I don't like any of the forEach changes. for in reads a lot better.

File: Sources/AudioKitEX/Sequencing/Sequencer.swift [hunk 1]
```diff
@@ -157,9 +159,7 @@ open class Sequencer {
 
     /// Remove all notes
     public func clear() {
-        for track in tracks {
-            track.clear()
-        }
+        tracks.forEach { $0.clear() }
```
Comment (aure) [inline Sources/AudioKitEX/Sequencing/Sequencer.swift:162]: This one is good to get on one line, still like for in better though.

**Review by aure — CHANGES_REQUESTED** [review]
To me, using forEach should be reserved for functional programming. These are just loops, not passing in a function which is what forEach can be good for.

**Comment by aure** [comment]
I suppose `forEach { track in` reads fine and gets rid of the `$0`. Here's a pretty reputable source for choosing between them. https://www.swiftbysundell.com/tips/picking-between-for-and-for-each/

**Review by aure — APPROVED** [review]

**Comment by aure** [comment]
The other place forEach is good is when you're dealing with optionals. I updated a bunch of AudioKit as well.

**Comment by eljeff** [comment]
I'm not married to the forEach method at all - either way is fine w me. I just saw it as a quick one-liner somewhere else in the code, so I erred on the side of brevity striving for consistency.

**Pull request merged**